packages/medusa/src/services/payment-provider.ts

```diff
@@ -0,0 +1,544 @@
+import { MedusaError } from "medusa-core-utils"
```

Comment:
question: should this have it's own pr?

Reply:
I regrouped all base refactoring in that PR because they all create a web, and that way I ve able to split the other plugins prs